inc/authors.php

```diff
@@ -93,3 +94,27 @@ function redirect_author_if_old_cap_prefix() : void {
     wp_safe_redirect( $redirect_url, 301 );
     exit();
 }
+
+/**
+ * Update post_author with the first co-author plus assigned author.
+ *
+ * @param array $data An array of slashed, sanitized, and processed post data.
+ * @param array $postarr An array of sanitized (and slashed) but otherwise unmodified post data.
+ *
+ * @return array Updated post data.
+ */
+function update_post_author( $data, $postarr ) {
+	if ( ! function_exists( 'get_coauthors' ) ) {
+		return $data;
+	}
+
+	$coauthors    = get_coauthors( $data['ID'] );
+	$author_login = str_replace( 'cap-', '', reset( $coauthors )->linked_account );
+	$author_id    = get_user_by( 'login', $author_login )->ID;
+
+	if ( ! wp_is_post_revision( $postarr['ID'] ) ) {
```

Comment:
You could move this check earlier to return early if its a revision, rather than looking up authors unnecessarily